crates/biome_js_analyze/src/semantic_analyzers/nursery/no_invalid_new_builtin.rs

```diff
@@ -0,0 +1,137 @@
+    ///
+    /// Several built-in objects cannot be instantiated and will throw a `TypeError`
+    /// if you try to execute them as constructors. These objects are:
+    ///
+    /// - [`Math`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Math)
+    /// - [`JSON`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/JSON)
+    /// - [`Reflect`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Reflect)
+    /// - [`Atomics`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Atomics)
+    /// - [`Intl`](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Intl)
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/no-new-native-nonconstructor/
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// var symbol = new Symbol('foo');
+    /// var bigInt = new BigInt(9007199254740991);
+    /// var math = new Math();
+    /// var json = new JSON();
+    /// var reflect = new Reflect();
+    /// var atomics = new Atomics();
+    /// var intl = new Intl();
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// var symbol = Symbol('foo');
+    /// var bigInt = BigInt(9007199254740991);
+    ///
+    /// // Ignores shadowed
+    /// function foo(Symbol) {
+    ///     const symbol = new Symbol('foo');
+    /// }
+    /// function bar(BigInt) {
+    ///     const bigInt = new BigInt(9007199254740991);
+    /// }
+    /// function baz(Math) {
+    ///     const math = new Math();
+    /// }
+    /// function qux(JSON) {
+    ///     const json = new JSON();
+    /// }
+    /// function quux(Reflect) {
+    ///     const reflect = new Reflect();
+    /// }
+    /// function corge(Intl) {
+    ///     const intl = new Intl();
+    /// }
+    /// ```
+    pub(crate) NoInvalidNewBuiltin {
+        version: "next",
+        name: "noInvalidNewBuiltin",
+        recommended: true,
+    }
+}
+
+impl Rule for NoInvalidNewBuiltin {
+    type Query = Semantic<JsNewExpression>;
+    type State = StaticValue;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let callee = ctx.query().callee().ok()?;
+        let (reference, name) = global_identifier(&callee)?;
+        match name.text() {
+            "Symbol" | "BigInt" | "Math" | "JSON" | "Reflect" | "Atomics" | "Intl" => {
+                ctx.model().binding(&reference).is_none().then_some(name)
+            }
+            _ => None,
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, builtin_name: &Self::State) -> Option<RuleDiagnostic> {
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            ctx.query().range(),
+            markup! {
+                <Emphasis>"`"{builtin_name.text()}"`"</Emphasis>" cannot be called as a constructor."
```

Comment:
We should explain **why** it can't be called as a construtor. Either here, or by adding a note.

Usually, we should follow this:
1. Call out the error
2. Tell why it's been a call out
3. Tell the user what they should do

Number 2 is currently missing